doc/feature/ReqRep.md

```diff
@@ -0,0 +1,112 @@
+Request-Reply {#reqrep}
+============
+
+Certain operations are by necessity synchronous, RPC-like exchanges. This
+specification describes this client-server API based on zmq req-rep (dealer/router) sockets.
+
+## Requirements
+
+* Asynchronous remote procedure call from a Client to a Server
+
+## API
+
+    using ReplyFunc = std::function<void(const void*, size_t)>;
+
+    class Client : public Receiver
```

Comment:
Is the inheritance from Receiver important from the user's point of view or is it just convenient for the implementation? If it's the latter replace it with implementetion-defined in the spec.